src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -0,0 +1,493 @@
+release type of `Flutter.framework` and `App.framework`.
+
+## Write code to use FlutterViewController from your host app
+
+The proper place to do this will be specific to your host app. Here is an
+example that makes sense for the blank screen of the host app generated
+by Xcode 10.0.
+
+First declare your app delegate to be a subclass of `FlutterAppDelegate`. Then define a FlutterEngine property, which help you to register a plugin without a FlutterViewController instance.
+
+In `AppDelegate.h`:
+
+```objective-c
+#import <UIKit/UIKit.h>
+#import <Flutter/Flutter.h>
+
+@interface AppDelegate : FlutterAppDelegate
+@property (nonatomic,strong) FlutterEngine *flutterEngine;
+@end
+```
+
+This allows `AppDelegate.m` to be really simple, unless your host app
+needs to override other methods here:
+
+```objective-c
+#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h> // Only if you have Flutter Plugins
+
+#include "AppDelegate.h"
+
+@implementation AppDelegate
+
+// This override can be omitted if you do not have any Flutter Plugins.
+- (BOOL)application:(UIApplication *)application
+    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  self.flutterEngine = [[FlutterEngine alloc] initWithName:@"io.flutter" project:nil];
+  [self.flutterEngine runWithEntrypoint:nil];
+  [GeneratedPluginRegistrant registerWithRegistry:self.flutterEngine];
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+@end
+```
+
+If you are writing in Swift, you can do the following in your `AppDelegate.swift`:
+
+```swift
+import UIKit
+import Flutter
+import FlutterPluginRegistrant // Only if you have Flutter Plugins.
+
+@UIApplicationMain
+class AppDelegate: FlutterAppDelegate {
+  var flutterEngine : FlutterEngine?;
+  // Only if you have Flutter plugins.
+  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    self.flutterEngine = FlutterEngine(name: "io.flutter", project: nil);
+    self.flutterEngine?.run(withEntrypoint: nil);
+    GeneratedPluginRegistrant.register(with: self.flutterEngine);
+    return super.application(application, didFinishLaunchingWithOptions: launchOptions);
+  }
+
+}
+```
+
+<details>
+<summary>What to do if the app delegate already inherits from somewhere else.</summary>
+
+Make your app delegate implement the `FlutterAppLifeCycleProvider` protocol, e.g.:
+
+```objective-c
+#import <Flutter/Flutter.h>
+#import <UIKit/UIKit.h>
+#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h> // Only if you have Flutter Plugins
+
+@interface AppDelegate : UIResponder <UIApplicationDelegate, FlutterAppLifeCycleProvider>
+@property (strong, nonatomic) UIWindow *window;
+@property (nonatomic,strong) FlutterEngine *flutterEngine;
+@end
+```
+
+The implementation should mostly just delegate to a `FlutterPluginAppLifeCycleDelegate`:
+
+```objective-c
+@implementation AppDelegate
+{
```

Comment:
We probably shouldn't encourage an ivar like this.
```objective-c
@interface AppDelegate ()
@property (readonly) FlutterPluginAppLifeCycleDelegate* lifeCycleDelegate;
@end
```
...
```objective-c
return [self.lifeCycleDelegate application:application didFinishLaunchingWithOptions:launchOptions];
```

Reply:
Do you have a style guide to reference?  That isn't specified in the google objective-c style guide.

Reply:
It's mentioned in the "You Can Define Instance Variables without Properties" section.
https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/EncapsulatingData/EncapsulatingData.html#//apple_ref/doc/uid/TP40011210-CH5-SW23

Reply:
done